### PR TITLE
Expose RpcClient's call_method function

### PR DESCRIPTION
**Pull request by valentinewallace**
For use in the sample node.

### CODE REVIEW

File: lightning-block-sync/src/convert.rs [hunk 1]
```diff
@@ -156,6 +156,14 @@ impl TryInto<(BlockHash, Option<u32>)> for JsonResponse {
 	}
 }
 
+impl TryInto<serde_json::Value> for JsonResponse {
```
Comment (TheBlueMatt) [inline lightning-block-sync/src/convert.rs:159]: Why TryInto instead of just making the inner field pub?

**Review by TheBlueMatt — COMMENTED** [review]
Concept ACK. Eventually hopefully we can drop this and move to rust-bitcoincore-rpc after https://github.com/rust-bitcoin/rust-bitcoincore-rpc/issues/166 (and maybe if we add rEST support there).

File: lightning-block-sync/src/convert.rs (hunk 1 above)
Comment (valentinewallace) [reply lightning-block-sync/src/convert.rs:159]: No reason. Is that better?

Comment (TheBlueMatt) [reply lightning-block-sync/src/convert.rs:159]: Just seems simpler, plus it avoids an unwrap that cannot fail at the callsite.

Comment (jkczyz) [reply lightning-block-sync/src/convert.rs:159]: `Into` is the non-failing version, but I agree making the inner field `pub` would be simpler.

File: lightning-block-sync/src/rpc.rs [hunk 2]
```diff
@@ -34,7 +34,7 @@ impl RpcClient {
 	}
 
 	/// Calls a method with the response encoded in JSON format and interpreted as type `T`.
-	async fn call_method<T>(&mut self, method: &str, params: &[serde_json::Value]) -> std::io::Result<T>
+	pub async fn call_method<T>(&mut self, method: &str, params: &[serde_json::Value]) -> std::io::Result<T>
```
Comment (jkczyz) [inline lightning-block-sync/src/rpc.rs:37]: Perhaps do the same for `RestClient` for consistency.

**Comment by jkczyz** [comment]
Superseded by #825 as discussed offline.

**Pull request closed without merging**